### PR TITLE
master: add `operate-source show`

### LINKED ISSUE #832: Add a way to check the sources
## Feature Request

**Is your feature request related to a problem? Please describe:**

After adding some sources. ref https://pingcap.com/docs-cn/tidb-data-migration/dev/manage-source/#%E5%8A%A0%E8%BD%BD%E6%95%B0%E6%8D%AE%E6%BA%90%E9%85%8D%E7%BD%AE
```
root@control:/tiup-cluster# dmctl --master-addr n1:8261 operate-source create ./_dev/dm/source1.toml
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "source": "mysql1",
            "worker": "dm-172.19.0.104-8262"
        }
    ]
}
```
there's no way to check the sources?
how can I know what sources I have later? And check some issues like if some sources have no worker?

**Describe the feature you'd like:**

Add a way to check the sources

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
fix #832

### What is changed and how it works?
`operate-source show` will output all source with their bounded worker (maybe empty)

### Check List 

Tests 

 - Integration test

Code changes

 - Has (user) interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: dm/ctl/master/operate_source.go [hunk 1]
```diff
@@ -85,6 +87,10 @@ func operateSourceFunc(cmd *cobra.Command, _ []string) {
 		common.PrintLines("invalid operate '%s' on worker", cmdType)
 		return
 	}
+	if op != pb.SourceOp_ShowSource && len(cmd.Flags().Args()) == 1 {
+		common.PrintLines("operate-source create/update/stop should input config-files")
```
Comment (csuzhangxc) [inline dm/ctl/master/operate_source.go:91]: `input` seems strange here, 😢

File: dm/master/server.go [hunk 2]
```diff
@@ -1426,7 +1426,12 @@ func (s *Server) OperateSource(ctx context.Context, req *pb.OperateSourceRequest
 		resp.Msg = err.Error()
 		return resp, nil
 	}
-	var workers []*scheduler.Worker
+
+	// sourceIDs and workers are used to query status from worker, to return a more real status
+	var (
+		sourceIDs []string
+		workers   []*scheduler.Worker
```
Comment (csuzhangxc) [inline dm/master/server.go:1433]: can we combine them into one `map`, like `sourceID --> worker`?

File: tests/dmctl_basic/run.sh [hunk 3]
```diff
@@ -131,6 +131,9 @@ function run() {
 
     dmctl_operate_source create $WORK_DIR/source1.yaml $SOURCE_ID1
     dmctl_operate_source create $WORK_DIR/source2.yaml $SOURCE_ID2
+    run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "operate-source show" \
```
Comment (csuzhangxc) [inline tests/dmctl_basic/run.sh:135]: How about showing detailed results both for _all sources bouned_ and _some sources not bouned_?

Comment (lance6716) [reply tests/dmctl_basic/run.sh:134]: https://github.com/pingcap/dm/pull/838/commits/2ba6fa215eaeeaa3897c33b8dca0c8b33b5e4308 added test

**Comment by csuzhangxc** [comment]
docs PR https://github.com/pingcap/docs-dm/pull/219

**Pull request merged**